pallene/checker.lua

```diff
@@ -465,6 +465,32 @@ function FunChecker:check_stat(stat)
             stat.decl._name = self.p.symbol_table:find_symbol(stat.decl.name)
             self:check_stat(stat.block)
         end)
+        
+    elseif tag == "ast.Stat.ForIn" then
+        local tab = self:check_exp_synthesize(stat.exp)
+
+        if tab.exp.var.name == "ipairs" then
```

Comment:
We need to add some error checking here:
* tab needs to be a Exp.CallFunc
* tab.exp needs to be an Exp.Var

Furthermore, it would be better to check if `var._name` points to the builtin `ipairs` function. (the code that checks `cname._tag` can give an idea of how this looks.